gcs_storage_client.go

```diff
@@ -390,3 +387,9 @@ func (util *snowflakeGcsClient) generateFileURL(stageLocation string, filename s
 func (util *snowflakeGcsClient) isTokenExpired(resp *http.Response) bool {
 	return resp.StatusCode == 401
 }
+
+func newGcsClient() gcsAPI {
```

Comment:
is this something GCS specific? Because it seems like the client is cloud provider agnostic, we are creating the same instance in s3_storage_client. Couldn't this method be renamed and  shared?

Reply:
Actually this is not that simple. Here we need to return `gcsAPI`, which is implemented by standard `http.Client`. We can't change it, because it would change the `client` variable type.